# **Project Name - Bird Species Observation Analysis**

**Project Type** - Exploratory Data Analysis
<br> **Contirbution** - Individual

# **Project Summary**


The Bird Species Observation Analysis project focuses on examining the diversity, distribution, and behavior of bird species across forest and grassland ecosystems using observational data collected from multiple administrative units. The primary goal is to clean, process, and analyze bird observation records to uncover ecological patterns influenced by habitat, seasonality, weather conditions, and observer behavior.

The project involves:

* Merging and cleaning multi-sheet Excel datasets

*   
Performing exploratory data analysis (EDA) to identify species trends
*  Studying correlations between environmental factors and bird activity


*   Creating interactive visualizations using Python and Streamlit
*   Deriving conservation insights from bird species distribution and behavior












# Git Hub Link -


https://github.com/San-antony2025/bird-observation-analysis

# **Problem Statement**

Bird populations are highly sensitive to changes in their environment. Understanding their behavior and habitat preferences is essential for biodiversity conservation and ecological planning. However, analyzing large-scale, multi-location bird observation data poses challenges such as inconsistent formats, missing values, and complex interdependencies between variables.

This project aims to:

* Analyze bird species distribution across forest vs. grassland habitats

* Understand the influence of environmental factors like temperature, wind, and humidity on bird activity

* Identify biodiversity hotspots, observer biases, and species at risk

* Uncover patterns in seasonal and geographic bird behaviors

# **Business Objective**

The findings from this project can support several real-world applications:



1.   **Wildlife Conservation**
 * Identify at-risk species and habitats requiring protection
 * Use species frequency and watchlist status for conservation planning


2. **Land and Habitat Management**
  * Guide ecological restoration and zoning based on species preference by habitat type
3. **Eco-Tourism Development**
  * Highlight bird-rich regions and seasons to promote nature-based tourism

4. **Sustainable Agriculture**

 * Inform farming practices that are compatible with bird habitat preservation

5. **Environmental Policy & Stewardship**

 * Provide data-backed evidence for decision-making in policy design and regional conservation efforts

6. **Biodiversity Monitoring**

 * Track long-term trends in bird diversity to assess ecosystem health and climate impacts
  





# **1.Know Your Data**

In [ ]:
import pandas as pd

# Raw GitHub links
forest_url = "https://raw.githubusercontent.com/San-antony2025/bird-observation-analysis/main/Bird_Monitoring_Data_FOREST.XLSX"
grassland_url = "https://raw.githubusercontent.com/San-antony2025/bird-observation-analysis/main/Bird_Monitoring_Data_GRASSLAND.XLSX"

# Read all sheets from each file
forest_sheets = pd.read_excel(forest_url, sheet_name=None)
grassland_sheets = pd.read_excel(grassland_url, sheet_name=None)

# Combine all forest sheets into one DataFrame
forest_df = pd.concat(forest_sheets.values(), ignore_index=True)

# Combine selected sheets from grassland
selected_sheets = ["ANTI", "HAFE", "MANA", "MONO"]
grassland_df = pd.concat([grassland_sheets[sheet] for sheet in selected_sheets if sheet in grassland_sheets], ignore_index=True)

# Merge both habitat datasets
combined_df = pd.concat([forest_df, grassland_df], ignore_index=True)

# Preview the combined dataset
print("Total rows:", combined_df.shape[0])
print("Total columns:", combined_df.shape[1])
combined_df.head()


Total rows: 17077
Total columns: 31


Admin_Unit_Code Sub_Unit_Code Site_Name  Plot_Name Location_Type  Year  \
0            ANTI           NaN    ANTI 1  ANTI-0036        Forest  2018   
1            ANTI           NaN    ANTI 1  ANTI-0036        Forest  2018   
2            ANTI           NaN    ANTI 1  ANTI-0036        Forest  2018   
3            ANTI           NaN    ANTI 1  ANTI-0036        Forest  2018   
4            ANTI           NaN    ANTI 1  ANTI-0036        Forest  2018   

        Date Start_Time  End_Time          Observer  ...  \
0 2018-05-22   06:19:00  06:29:00  Elizabeth Oswald  ...   
1 2018-05-22   06:19:00  06:29:00  Elizabeth Oswald  ...   
2 2018-05-22   06:19:00  06:29:00  Elizabeth Oswald  ...   
3 2018-05-22   06:19:00  06:29:00  Elizabeth Oswald  ...   
4 2018-05-22   06:19:00  06:29:00  Elizabeth Oswald  ...   

   PIF_Watchlist_Status Regional_Stewardship_Status Temperature   Humidity  \
0                 False                        True        19.9  79.400002   
1                 False                       False        19.9  79.400002   
2                 False                       False        19.9  79.400002   
3                 False                       False        19.9  79.400002   
4                 False                       False        19.9  79.400002   

               Sky                                   Wind         Disturbance  \
0  Cloudy/Overcast  Calm (< 1 mph) smoke rises vertically  No effect on count   
1  Cloudy/Overcast  Calm (< 1 mph) smoke rises vertically  No effect on count   
2  Cloudy/Overcast  Calm (< 1 mph) smoke rises vertically  No effect on count   
3  Cloudy/Overcast  Calm (< 1 mph) smoke rises vertically  No effect on count   
4  Cloudy/Overcast  Calm (< 1 mph) smoke rises vertically  No effect on count   

  Initial_Three_Min_Cnt  TaxonCode  Previously_Obs  
0                  True        NaN             NaN  
1                  True        NaN             NaN  
2                 False        NaN             NaN  
3                 False        NaN             NaN  
4                 False        NaN             NaN  

[5 rows x 31 columns]

# **2.Data Cleaning & Pre-processing**

In [ ]:
#checking shape, rows, columns
print(combined_df.shape)
print(combined_df.columns)
combined_df.info()


(17077, 31)
Index(['Admin_Unit_Code', 'Sub_Unit_Code', 'Site_Name', 'Plot_Name',
       'Location_Type', 'Year', 'Date', 'Start_Time', 'End_Time', 'Observer',
       'Visit', 'Interval_Length', 'ID_Method', 'Distance', 'Flyover_Observed',
       'Sex', 'Common_Name', 'Scientific_Name', 'AcceptedTSN', 'NPSTaxonCode',
       'AOU_Code', 'PIF_Watchlist_Status', 'Regional_Stewardship_Status',
       'Temperature', 'Humidity', 'Sky', 'Wind', 'Disturbance',
       'Initial_Three_Min_Cnt', 'TaxonCode', 'Previously_Obs'],
      dtype='object')
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17077 entries, 0 to 17076
Data columns (total 31 columns):
 #   Column                       Non-Null Count  Dtype         
---  ------                       --------------  -----         
 0   Admin_Unit_Code              17077 non-null  object        
 1   Sub_Unit_Code                722 non-null    object        
 2   Site_Name                    8546 non-null   object        
 3   Plot_Name          

In [ ]:
#checking columns with missing values
combined_df.isnull().sum().sort_values(ascending=False)




Sub_Unit_Code                  16355
TaxonCode                       8548
Previously_Obs                  8546
Site_Name                       8531
NPSTaxonCode                    8531
Sex                             5183
Distance                        1486
AcceptedTSN                       33
ID_Method                          2
Start_Time                         0
Year                               0
Date                               0
Location_Type                      0
Plot_Name                          0
Admin_Unit_Code                    0
Flyover_Observed                   0
Interval_Length                    0
Observer                           0
Visit                              0
End_Time                           0
AOU_Code                           0
Scientific_Name                    0
Common_Name                        0
Regional_Stewardship_Status        0
PIF_Watchlist_Status               0
Temperature                        0
Humidity                           0
Wind                               0
Sky                                0
Initial_Three_Min_Cnt              0
Disturbance                        0
dtype: int64

In [ ]:
# Drop columns with too many missing values
columns_to_drop = ['Sub_Unit_Code', 'TaxonCode', 'Previously_Obs', 'Site_Name', 'NPSTaxonCode']
existing_columns_to_drop = [col for col in columns_to_drop if col in combined_df.columns]
combined_df = combined_df.drop(columns=existing_columns_to_drop)

# Fill missing values in key columns
combined_df['Sex'] = combined_df['Sex'].fillna('Undetermined')
combined_df['Distance'] = combined_df['Distance'].fillna('Unknown')
combined_df['AcceptedTSN'] = combined_df['AcceptedTSN'].fillna(-1)
combined_df['ID_Method'] = combined_df['ID_Method'].fillna('Unknown')



In [ ]:
# Converting date and time columns

combined_df['Date'] = pd.to_datetime(combined_df['Date'], errors='coerce')
combined_df['Start_Time'] = pd.to_datetime(combined_df['Start_Time'], format='%H:%M:%S', errors='coerce').dt.time
combined_df['End_Time'] = pd.to_datetime(combined_df['End_Time'], format='%H:%M:%S', errors='coerce').dt.time


In [ ]:
#Removing dupilicates if any

combined_df = combined_df.drop_duplicates()


In [ ]:
# Creating new columns- seasons

def get_season(month):
    if month in [3, 4, 5]:
        return 'Spring'
    elif month in [6, 7, 8]:
        return 'Summer'
    elif month in [9, 10, 11]:
        return 'Fall'
    else:
        return 'Winter'

combined_df['Season'] = combined_df['Date'].dt.month.apply(get_season)


#viewing the created column
print(combined_df['Season'].value_counts())



Season
Summer    10505
Spring     4863
Name: count, dtype: int64


In [ ]:
# Creating New column - Observation minutes

#combine with date to form full datetime
combined_df['StartDateTime'] = pd.to_datetime(combined_df['Date'].astype(str) + ' ' + combined_df['Start_Time'].astype(str))
combined_df['EndDateTime'] = pd.to_datetime(combined_df['Date'].astype(str) + ' ' + combined_df['End_Time'].astype(str))

# making the calculation
combined_df['Observation_Duration'] = combined_df['EndDateTime'] - combined_df['StartDateTime']
combined_df['Observation_Duration_Min'] = combined_df['Observation_Duration'].dt.total_seconds() / 60

#to view the created column
print(combined_df[['Start_Time', 'End_Time', 'StartDateTime', 'EndDateTime', 'Observation_Duration_Min']].head())



  Start_Time  End_Time       StartDateTime         EndDateTime  \
0   06:19:00  06:29:00 2018-05-22 06:19:00 2018-05-22 06:29:00   
1   06:19:00  06:29:00 2018-05-22 06:19:00 2018-05-22 06:29:00   
2   06:19:00  06:29:00 2018-05-22 06:19:00 2018-05-22 06:29:00   
3   06:19:00  06:29:00 2018-05-22 06:19:00 2018-05-22 06:29:00   
4   06:19:00  06:29:00 2018-05-22 06:19:00 2018-05-22 06:29:00   

   Observation_Duration_Min  
0                      10.0  
1                      10.0  
2                      10.0  
3                      10.0  
4                      10.0  


In [ ]:
# Saving the cleaned data

combined_df.to_csv("cleaned_bird_data.csv", index=False)
print("Total rows in cleaned data:", len(combined_df))
display(combined_df.head())


Total rows in cleaned data: 15368


Admin_Unit_Code  Plot_Name Location_Type  Year       Date Start_Time  \
0            ANTI  ANTI-0036        Forest  2018 2018-05-22   06:19:00   
1            ANTI  ANTI-0036        Forest  2018 2018-05-22   06:19:00   
2            ANTI  ANTI-0036        Forest  2018 2018-05-22   06:19:00   
3            ANTI  ANTI-0036        Forest  2018 2018-05-22   06:19:00   
4            ANTI  ANTI-0036        Forest  2018 2018-05-22   06:19:00   

   End_Time          Observer  Visit Interval_Length  ...   Humidity  \
0  06:29:00  Elizabeth Oswald      1       0-2.5 min  ...  79.400002   
1  06:29:00  Elizabeth Oswald      1       0-2.5 min  ...  79.400002   
2  06:29:00  Elizabeth Oswald      1     2.5 - 5 min  ...  79.400002   
3  06:29:00  Elizabeth Oswald      1     2.5 - 5 min  ...  79.400002   
4  06:29:00  Elizabeth Oswald      1     2.5 - 5 min  ...  79.400002   

               Sky                                   Wind         Disturbance  \
0  Cloudy/Overcast  Calm (< 1 mph) smoke rises vertically  No effect on count   
1  Cloudy/Overcast  Calm (< 1 mph) smoke rises vertically  No effect on count   
2  Cloudy/Overcast  Calm (< 1 mph) smoke rises vertically  No effect on count   
3  Cloudy/Overcast  Calm (< 1 mph) smoke rises vertically  No effect on count   
4  Cloudy/Overcast  Calm (< 1 mph) smoke rises vertically  No effect on count   

  Initial_Three_Min_Cnt  Season       StartDateTime         EndDateTime  \
0                  True  Spring 2018-05-22 06:19:00 2018-05-22 06:29:00   
1                  True  Spring 2018-05-22 06:19:00 2018-05-22 06:29:00   
2                 False  Spring 2018-05-22 06:19:00 2018-05-22 06:29:00   
3                 False  Spring 2018-05-22 06:19:00 2018-05-22 06:29:00   
4                 False  Spring 2018-05-22 06:19:00 2018-05-22 06:29:00   

   Observation_Duration  Observation_Duration_Min  
0       0 days 00:10:00                      10.0  
1       0 days 00:10:00                      10.0  
2       0 days 00:10:00                      10.0  
3       0 days 00:10:00                      10.0  
4       0 days 00:10:00                      10.0  

[5 rows x 31 columns]

# **3. Exploratory Data Analysis**

## **1. Temporal Analysis**

**1.1: Observation by year,month,season**

In [ ]:

# Extract year, month, season
combined_df['Year'] = combined_df['Date'].dt.year
combined_df['Month'] = combined_df['Date'].dt.month

def get_season(month):
    if month in [3, 4, 5]:
        return 'Spring'
    elif month in [6, 7, 8]:
        return 'Summer'
    elif month in [9, 10, 11]:
        return 'Fall'
    else:
        return 'Winter'

combined_df['Season'] = combined_df['Month'].apply(get_season)

# Observation frequency
print(combined_df['Year'].value_counts().sort_index()) # Check the number of observations per year to understand the temporal range
print(combined_df['Month'].value_counts().sort_index()) # Check observation frequency per month to identify survey intensity or seasonal patterns
print(combined_df['Season'].value_counts()) # Map months to seasons and count how many observations fall in each season


Year
2018    15368
Name: count, dtype: int64
Month
5    4863
6    6209
7    4296
Name: count, dtype: int64
Season
Summer    10505
Spring     4863
Name: count, dtype: int64


**1.2: Observation Time**

In [ ]:

# Count number of observations per hour (based on Start_Time) to analyze bird activity by time of day
combined_df['Hour'] = pd.to_datetime(combined_df['Start_Time'].astype(str), errors='coerce').dt.hour
combined_df['Hour'].value_counts().sort_index()



/tmp/ipython-input-1708291786.py:2: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  combined_df['Hour'] = pd.to_datetime(combined_df['Start_Time'].astype(str), errors='coerce').dt.hour


Hour
5     1288
6     3837
7     4014
8     3194
9     2014
10    1021
Name: count, dtype: int64

# **2. Spatial Analysis**

**2.1: Species Diversity by Habitat Type**

In [ ]:
# Count the number of unique species observed in each habitat type (e.g., Forest vs Grassland)
species_by_habitat = combined_df.groupby('Location_Type')['Scientific_Name'].nunique()
print(species_by_habitat)




Location_Type
Forest       108
Grassland    107
Name: Scientific_Name, dtype: int64


**2.2: Species Diversity by Administrative Unit (Park)**

In [ ]:
# Count the number of unique species in each administrative unit (e.g., ANTI, MONO, etc.)
species_by_unit = combined_df.groupby('Admin_Unit_Code')['Scientific_Name'].nunique().sort_values(ascending=False)
print(species_by_unit.head(10))  # Top 10 units






Admin_Unit_Code
MONO    100
MANA     81
ANTI     81
CHOH     80
NACE     66
HAFE     55
PRWI     54
GWMP     49
CATO     46
ROCR     45
Name: Scientific_Name, dtype: int64


**2.3: Species Diversity by Plot**

In [ ]:
# Count the number of unique species observed in each survey plot
species_by_plot = combined_df.groupby('Plot_Name')['Scientific_Name'].nunique().sort_values(ascending=False)
print(species_by_plot.head(10))  # Top 10 plots


Plot_Name
ANTI-0105    27
MANA-0047    27
MONO-0057    27
MONO-0066    26
CHOH-0812    26
MONO-0076    26
MONO-0085    26
MANA-0048    26
ANTI-0160    25
ANTI-0034    25
Name: Scientific_Name, dtype: int64


# **3. Species Analysis**

**3.1: Total bird species Diversity**

In [ ]:
# Count the total number of unique bird species observed across all locations
total_unique_species = combined_df['Scientific_Name'].nunique()
print("Total Unique Species:", total_unique_species)


Total Unique Species: 127


**3.2: Most Commonly Observed Species**

In [ ]:
# List the top 10 most frequently observed bird species (by common name)
top_species = combined_df['Common_Name'].value_counts().head(10)
print("Top 10 Most Observed Species:\n", top_species)


Top 10 Most Observed Species:
 Common_Name
Northern Cardinal          1125
Carolina Wren               993
Red-eyed Vireo              737
Eastern Tufted Titmouse     720
Indigo Bunting              611
Eastern Wood-Pewee          574
Field Sparrow               492
Red-bellied Woodpecker      488
American Robin              470
Acadian Flycatcher          462
Name: count, dtype: int64


**3.3: Observation Activity Type (ID Method)**

In [ ]:
# Count how many times each ID method was used (e.g., Singing, Calling, Visualization)
activity_methods = combined_df['ID_Method'].value_counts()
print("Observation Activity Types:\n", activity_methods)


Observation Activity Types:
 ID_Method
Singing          9620
Calling          3940
Visualization    1806
Unknown             2
Name: count, dtype: int64


**3.4: Observation Time Duration**

In [ ]:
# Display unique interval lengths to see how long observations typically lasted
intervals = combined_df['Interval_Length'].value_counts()
print("Observation Durations:\n", intervals)


Observation Durations:
 Interval_Length
0-2.5 min       7752
2.5 - 5 min     3149
5 - 7.5 min     2386
7.5 - 10 min    2081
Name: count, dtype: int64


**3.5: Sex Distribution**

In [ ]:
# Count how many observations were marked male, female, or undetermined
sex_ratio = combined_df['Sex'].value_counts()
print("Sex Ratio:\n", sex_ratio)


Sex Ratio:
 Sex
Undetermined    12133
Male             3109
Female            126
Name: count, dtype: int64


#**4. Environmental condition**

**4.1: Summary of Temperature & Humidity**

In [ ]:
# Display basic statistics (mean, min, max, etc.) for temperature and humidity
print("Temperature Summary:\n", combined_df['Temperature'].describe())
print("Humidity Summary:\n", combined_df['Humidity'].describe())


Temperature Summary:
 count    15368.000000
mean        22.493285
std          4.193385
min         11.000000
25%         19.700001
50%         22.200001
75%         25.000000
max         37.299999
Name: Temperature, dtype: float64
Humidity Summary:
 count    15368.000000
mean        74.157763
std         12.169331
min          7.300000
25%         68.000000
50%         76.599998
75%         83.325003
max         98.800003
Name: Humidity, dtype: float64


**4.2: Sky Conditions**

In [ ]:
# Count how many observations were made under each sky condition (e.g., clear, cloudy)
sky_conditions = combined_df['Sky'].value_counts()
print("Sky Conditions:\n", sky_conditions)


Sky Conditions:
 Sky
Partly Cloudy          6172
Clear or Few Clouds    5330
Cloudy/Overcast        2916
Fog                     598
Mist/Drizzle            352
Name: count, dtype: int64


**4.3: Wind Conditions**

In [ ]:
# Count wind condition categories reported during bird observations
wind_conditions = combined_df['Wind'].value_counts()
print("Wind Conditions:\n", wind_conditions)


Wind Conditions:
 Wind
Light air movement (1-3 mph) smoke drifts     7635
Calm (< 1 mph) smoke rises vertically         4210
Light breeze (4-7 mph) wind felt on face      3159
Gentle breeze (8-12 mph), leaves in motion     364
Name: count, dtype: int64


**4.4: Disturbance Effects**

In [ ]:
# Assess how many observations recorded disturbance levels (e.g., no effect, slight effect)
disturbance_impact = combined_df['Disturbance'].value_counts()
print("Disturbance Effects:\n", disturbance_impact)


Disturbance Effects:
 Disturbance
No effect on count          7525
Slight effect on count      5836
Moderate effect on count    1577
Serious effect on count      430
Name: count, dtype: int64


# **5. Distance & Flyover Behavior**

**5.1: Distance Distribution**

In [ ]:
# Count how far birds were typically observed from the observer
distance_distribution = combined_df['Distance'].value_counts()
print("Distance Distribution:\n", distance_distribution)


Distance Distribution:
 Distance
50 - 100 Meters    7774
<= 50 Meters       6905
Unknown             689
Name: count, dtype: int64


**5.2: Flyover Behavior**

In [ ]:
# Count how often birds were flying over the observer during observation
flyover_counts = combined_df['Flyover_Observed'].value_counts()
print("Flyover Observations:\n", flyover_counts)


Flyover Observations:
 Flyover_Observed
False    14679
True       689
Name: count, dtype: int64


# **6. Observer Trends**

**6.1: Most Active Observers**

In [ ]:
# Identify observers with the highest number of records
top_observers = combined_df['Observer'].value_counts().head(10)
print("Top Observers:\n", top_observers)


Top Observers:
 Observer
Elizabeth Oswald    5759
Kimberly Serno      5346
Brian Swimelar      4263
Name: count, dtype: int64


**6.2: Visit Impact on Species Count**

In [ ]:
# Analyze how species diversity changes across different visits
visit_effect = combined_df.groupby('Visit')['Scientific_Name'].nunique().sort_index()
print("Species Diversity by Visit Number:\n", visit_effect)


Species Diversity by Visit Number:
 Visit
1    119
2    100
3     74
Name: Scientific_Name, dtype: int64


# **7. Conservation Insights**

**7.1: Species on the PIF Watchlist**

In [37]:
# List most frequently observed species that are flagged as conservation watchlist species
watchlist_species = combined_df[combined_df['PIF_Watchlist_Status'] == True]['Scientific_Name'].value_counts().head(10)
print("PIF Watchlist Species:\n", watchlist_species)


PIF Watchlist Species:
 Scientific_Name
Hylocichla mustelina          309
Helmitheros vermivorus         31
Setophaga discolor             25
Setophaga cerulea               7
Oporornis formosus              2
Empidonax traillii              2
Vermivora cyanoptera            1
Melanerpes erythrocephalus      1
Name: count, dtype: int64


**7.2: Regional Stewardship Species**

In [38]:
# List species that have high regional stewardship importance
steward_species = combined_df[combined_df['Regional_Stewardship_Status'] == True]['Scientific_Name'].value_counts().head(10)
print("Regional Stewardship Species:\n", steward_species)


Regional Stewardship Species:
 Scientific_Name
Passerina cyanea           611
Contopus virens            574
Spizella pusilla           492
Empidonax virescens        462
Poecile carolinensis       365
Hylocichla mustelina       309
Picoides pubescens         260
Pipilo erythrophthalmus    257
Piranga olivacea           216
Setophaga citrina           64
Name: count, dtype: int64


**7.3:Most Common AOU Codes**

In [39]:
# Display the top AOU codes based on observation frequency
aou_top = combined_df['AOU_Code'].value_counts().head(10)
print("Top AOU Codes:\n", aou_top)


Top AOU Codes:
 AOU_Code
NOCA    1125
CARW     993
REVI     737
ETTI     720
INBU     611
EAWP     574
FISP     492
RBWO     488
AMRO     470
ACFL     462
Name: count, dtype: int64


## **Manupulation Done & Insights Found**


# **Manupulations Done**

1. **Data Cleaning**


*   Removed columns with too many missing values (TaxonCode,Sub_Unit_Code, Site_Name, etc.)

*   Filled missing categorical values:


           *   Sex → filled with 'Undetermined'   
           *   Distance, ID_Method → filled with 'Unknown'
*   Converted Date, Start_Time, and End_Time into proper datetime formats.

🔹 2. **Feature Engineering**



*  Extracted Year and Month from Date
*  Created Season based on month (Spring, Summer, Fall, Winter)
*  Extracted Hour from Start_Time for time-of-day analysis
* Combined Forest and Grassland sheets into a single cleaned DataFrame


## **Insights Found**


**1. Temporal Insights**

* All observations are from 2018

* Most activity recorded in June, during Summer

* Peak bird activity: 6–8 AM, especially 7 AM

**2. Spatial Insights**

* Forest and Grassland have nearly equal species diversity
* Most diverse unit: MONO (100 species)
* Top plots: MONO-0057,0066, MANA 0047,ANTI 0105 with 27–26 species

**3. Species Insights**

* Total unique species: 127

* Most detections via Singing and Calling
* Many birds marked as ‘Undetermined’ sex

**4. Environmental Insights**

* Avg Temp: 22.2°C, Humidity: 76.5%
* Mostly Clear or Partly Cloudy skies
* Disturbance mostly reported as ‘No effect’

**5.  Distance & Behavior**

* Most birds observed within 50-100 meters

* Flyovers were rare

**6. Observer Trends**

* Top contributors: Elizabeth Oswald, Kimberly Serno

* Visit 1 had highest species diversity

**7. Conservation Insights**

* Watchlist and Stewardship species like Hylocichla mustelina, Passerina cyanea were observed

* Frequent AOU codes: NOCA, CARW, REVI



# **4. Visualisation**

The dashboard built with Streamlit + Plotly allows:
- Species-level filtering
- Monthly and seasonal trend visualization
- Heatmap exploration of temporal density
- Environmental correlation charts

Deployment Link: (https://bird-observation-analysis-ff6t6oks43qpkypq6sx2qp.streamlit.app/)

# **5.Conclusion**

This project focuses on analyzing bird observation data collected from multiple administrative units across forest and grassland habitats. The dataset includes detailed records of species sightings, observation times, environmental conditions, and observer details.

The analysis reveals that habitat type, time of day, and environmental conditions play critical roles in bird sightings.
By combining data-driven insights with interactive visualization, this project enables informed decision-making for biodiversity conservation and ecological research.